src/main/java/com/google/devtools/build/lib/actions/Actions.java

```diff
@@ -371,7 +373,19 @@ public static String escapedPath(String path) {
    * that no other label maps to this string.
    */
   public static String escapeLabel(Label label) {
-    return PATH_ESCAPER.escape(label.getPackageName() + ":" + label.getName());
+    String path = label.getPackageName() + ":" + label.getName();
+    if (!label.getRepository().isMain()) {
```

Comment:
Are there any alternative solutions?

I know of an attempt by TF team to demangle the library in @rules_cc/cc/private/toolchain/osx_cc_wrapper.sh.tpl

Using one way hash would prevent that.

Reply:
We can use the name as is and wait for but reports that may never come 🙂

Just to understand this better, why would they need to demangle the name?

Reply:
I think it’s to compile on RBE and run locally.

Reply:
I implemented a reversible scheme, PTAL.